cli/tools/test_runner.rs

```diff
@@ -199,8 +199,128 @@ impl TestReporter for PrettyTestReporter {
   }
 }
 
-fn create_reporter(concurrent: bool) -> Box<dyn TestReporter + Send> {
-  Box::new(PrettyTestReporter::new(concurrent))
+struct TerseTestReporter {
+  time: Instant,
+  failed: usize,
+  filtered_out: usize,
+  ignored: usize,
+  passed: usize,
+  measured: usize,
+  pending: usize,
+  failures: Vec<(String, String)>,
+}
+
+impl TerseTestReporter {
+  fn new() -> TerseTestReporter {
+    TerseTestReporter {
+      time: Instant::now(),
+      failed: 0,
+      filtered_out: 0,
+      ignored: 0,
+      passed: 0,
+      measured: 0,
+      pending: 0,
+      failures: Vec::new(),
+    }
+  }
+}
+
+impl TestReporter for TerseTestReporter {
+  fn visit_event(&mut self, event: TestEvent) {
+    match &event.message {
+      TestMessage::Plan {
+        pending,
+        filtered,
+        only: _,
+      } => {
+        if *pending == 1 {
+          println!("running {} test from {}", pending, event.origin);
+        } else {
+          println!("running {} tests from {}", pending, event.origin);
+        }
+
+        self.pending += pending;
+        self.filtered_out += filtered;
+      }
+
+      TestMessage::Result {
+        name,
+        duration: _,
+        result,
+      } => {
+        self.pending -= 1;
+
+        match result {
+          TestResult::Ok => {
+            print!("{}", colors::green("."),);
+
+            self.passed += 1;
+          }
+          TestResult::Ignored => {
+            print!("{}", colors::yellow("i"),);
+
+            self.ignored += 1;
+          }
+          TestResult::Failed(error) => {
+            print!("{}", colors::red("F"),);
+
+            self.failed += 1;
+            self.failures.push((name.to_string(), error.to_string()));
+          }
+        }
+      }
+
+      _ => {}
+    }
+  }
+
+  fn done(&mut self) {
+    if !self.failures.is_empty() {
```

Comment:
Little bit of duplication here, will follow up and make these reporters stateless.